### PR TITLE
Intermediate of a conversation added successfully.

### PR DESCRIPTION
**Pull request by SinghHimadri**
## Issue that this pull request solves
#87 
Closes: # (issue number)
#87 
## Proposed changes

Brief description of what is fixed or changed

One person can talk to another person from your WhatsApp!
Use command python3 -m -wi -s (sender) -r  (receiver)
Person A: send message to you
You: message received from A and now the user needs to type #_FWD and the last message received from A will be forwarded to Person B
Person B: receives message from you
I have used function objective to access the last message.

## Types of changes

_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation update (Documentation content changed)
- [ ] Other (please describe): 

## Checklist

_Put an `x` in the boxes that apply_

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

## Screenshots

Please attach the screenshots of the changes made in case of change in user interface
Terminal
Screenshot from 2020-05-25 16-13-28

Messages received from person A

WhatsApp Image 2020-05-22 at 3 54 14 PM

Messaged forwarded to person B

WhatsApp Image 2020-05-22 at 3 54 14 PM (1)

## Other information

Any other information that is important to this pull request

### CODE REVIEW

**Comment by rpotter12** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- wplay/terminal_chat.py  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5558230&bid=18229749)

**Comment by SinghHimadri** [comment]
@rpotter12 Please review my PR.

**Comment by priyansh19** [comment]
Your changes look's great @SinghHimadri Just make the small changes mentioned by @rpotter12 as well :)

**Comment by rpotter12** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- wplay/chat_intermediator.py  1
- wplay/terminal_chat.py  1
- wplay/__main__.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5558230&bid=18229749)

**Comment by SinghHimadri** [comment]
> @SinghHimadri You should add an another option for this feature in which it ask for 2 command line argument for receiver and sender and rest they can chat by having you as a intermediator which using any `#_FWD` command

I have done the changes you asked. I hope this what it meant :) I have changed the description and photos please see.

**Comment by rpotter12** [comment]
@SinghHimadri I mean to say to insert a command like: `wplay -wi <sender> <receiver>`.

**Comment by SinghHimadri** [comment]
> @SinghHimadri I mean to say to insert a command like: `wplay -wi <contact1> <contact2>`. This will be more specific. As we are creating an intermediator so it should be from both the side: contact1 can send message to contact2 and vice versa.

Initially I tried to  do this `wplay -wi <contact1> <contact2>` but it has some issues like contact1 as first name and contacts as last name and takes it as one target and contacts as none.And also if some user wants to enter a target with first name and last name that's why I had to add parser for sender and receiver.

**Comment by rpotter12** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- wplay/__main__.py  1
- wplay/chat_intermediator.py  1
- wplay/utils/io.py  1
- wplay/terminal_chat.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5558230&bid=18229749)

**Comment by rpotter12** [comment]
> Initially, I tried to do this `wplay -wi <contact1> <contact2>` but it has some issues like it takes contact1 as first name and contact2 as the last name and takes it as one target and contact2 as none. And also if some user wants to enter a target with first name and last name that's why I had to add parser for sender and receiver. In the issue, it was mentioned that person A sends the message and it is sent to person B. In vice versa, it'll be difficult to maintain as at a time one chat is open i.e. one target (and also I think it's a different issue). I have completed what was mentioned in the issue.

Did you have the command like this: `wplay -wi "<contact1>" "<contact2>".
by using `"___"` it will take 2 arguments separately.

**Comment by SinghHimadri** [comment]
> > Initially, I tried to do this `wplay -wi <contact1> <contact2>` but it has some issues like it takes contact1 as first name and contact2 as the last name and takes it as one target and contact2 as none. And also if some user wants to enter a target with first name and last name that's why I had to add parser for sender and receiver. In the issue, it was mentioned that person A sends the message and it is sent to person B. In vice versa, it'll be difficult to maintain as at a time one chat is open i.e. one target (and also I think it's a different issue). I have completed what was mentioned in the issue.
> 
> Did you have the command like this: `wplay -wi "<contact1>" "<contact2>". by using `"___"` it will take 2 arguments separately.

Can you please guide me I'm not able to implement this.

**Comment by rpotter12** [comment]
@SinghHimadri https://www.tutorialspoint.com/python/python_command_line_arguments.htm
Maybe this can help to you

File: wplay/save_chat.py [hunk 1]
```diff
@@ -246,4 +244,4 @@ def runMain(mode, asset, bID):
             if len(drives) > 1:
                 print('Backup: ' + str(i))
                 folder = 'WhatsApp-' + str(i)
-            getMultipleFiles(drive[1], folder)
\ No newline at end of file
```
Comment (rpotter12) [inline wplay/save_chat.py:249]: Codacy Issue found: [String statement has no effect](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5558230)

**Comment by rpotter12** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- wplay/__main__.py  1
- wplay/chat_intermediator.py  1
- wplay/terminal_chat.py  1
- wplay/utils/io.py  1
         

Clones added
============
- wplay/save_chat.py  3
- wplay/terminal_chat.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5558230&bid=18229749)

**Comment by rpotter12** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- wplay/terminal_chat.py  1
- wplay/utils/io.py  1
- wplay/__main__.py  1
- wplay/chat_intermediator.py  1
         

Clones added
============
- wplay/terminal_chat.py  1
- wplay/save_chat.py  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5558230&bid=18229749)

File: wplay/save_chat.py [hunk 2]
```diff
@@ -246,4 +244,5 @@ def runMain(mode, asset, bID):
...
+            getMultipleFiles(drive[1], folder)
```
Comment (rpotter12) [inline wplay/save_chat.py:247]: Codacy Issue found: [String statement has no effect](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5558230)

**Comment by rpotter12** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- wplay/terminal_chat.py  1
- wplay/__main__.py  1
- wplay/chat_intermediator.py  1
- wplay/utils/io.py  1
         

Clones added
============
- wplay/terminal_chat.py  1
- wplay/save_chat.py  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5558230&bid=18229749)

**Comment by SinghHimadri** [comment]
@rpotter12  I have added the command line arguments not the user can input the command : ' -m play -wi "contact 1" "contacted".Also I have removed some error in about_change file and EOL in save chat.

**Comment by rpotter12** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- wplay/utils/io.py  1
- wplay/__main__.py  1
- wplay/chat_intermediator.py  1
         

Clones added
============
- wplay/terminal_chat.py  1
- wplay/save_chat.py  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5558230&bid=18229749)

**Comment by rpotter12** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- wplay/utils/io.py  1
- wplay/__main__.py  1
- wplay/terminal_chat.py  1
- wplay/chat_intermediator.py  1
         

Clones added
============
- wplay/terminal_chat.py  1
- wplay/save_chat.py  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5558230&bid=18229749)

**Comment by SinghHimadri** [comment]
@rpotter12  I have resolved all conflicts and done the changes you asked for.

**Comment by rpotter12** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- wplay/utils/io.py  1
- wplay/terminal_chat.py  1
- wplay/__main__.py  1
- wplay/chat_intermediator.py  1
         

Clones added
============
- wplay/terminal_chat.py  1
- wplay/save_chat.py  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5558230&bid=18229749)

**Comment by SinghHimadri** [comment]
> @SinghHimadri Please delete all the commented portion from save_chat.py

Done

**Comment by SinghHimadri** [comment]
> > @SinghHimadri Please delete all the commented portion from save_chat.py
> 
> Done

**Pull request merged**